### DISCUSSION TITLE
how to implement NLA authentication for Freerdp proxy?

### DISCUSSION THREAD

**Discussion by BinaryRyan** [FreeRDP/FreeRDP#10809] [Q&A]
Hi,I am using the freerdp proxy module.
I found xfreerdp cannot connect to freerdp-proxy when I did not specified /sec:tls.I guess freerdp server do not supoort NLA.But what puzzles me is why xfreerdp doesn't automatically negotiate encryption components just like mstsc. Or did the server close the connection directly after encountering NLA authentication,Why not continue negotiating other encryption methods？
Is there any other way to solve this problem?

## /sec:nla  or  do not proactively specify any encryption method

![QQ_1730451758322](https://github.com/user-attachments/assets/d4a14d6d-3c23-4914-9825-520057dd5dd5)
Looking forward to your reply

**Comment by akallabeth** [comment]
* configure your proxy correctly. `mstsc` just does a secondary connection with `/sec:tls` if `/sec:nla` fails (which we decided not to mimic) - so best deactivate `NLA` for the proxy (you can keep it on for the `proxy -> target server` authentication)
* If you want to support `NLA` for the `client -> proxy` authentication you need to implement it yourself. `NLA` does not transmit the credentials and the proxy needs those to authenticate to the target server. So you need some mapping on the proxy (`SAM` file for `client -> proxy` authentication along with a password database for the `proxy -> target server` authentication or you implement full kerberos - which I assume would be overkill)
